maven-core/src/main/java/org/apache/maven/project/ProjectModelResolver.java

```diff
@@ -285,6 +275,16 @@ public ModelSource resolveModel( final Dependency dependency )
             }
 
             dependency.setVersion( versionRangeResult.getHighestVersion().toString() );
+            
+            if ( modelPool != null )
+            {
+                Model model = Optional.ofNullable( modelPool.get( dependency.getGroupId(), dependency.getArtifactId(),
```

Comment:
`Optional.ofNullable(....).orElse(null)` seems like a code smell to me.

Reply:
Good catch, this line of code has moved a lot, including refactoring. Indeed no need for the optional anymore